src/java/org/apache/cassandra/gms/IFailureDetector.java

```diff
@@ -37,6 +40,28 @@ public interface IFailureDetector
      */
     public boolean isAlive(InetAddressAndPort ep);
 
+    /**
+     * Sorts the endpoints based on their response time.
+     * Nodes at smaller indexes have faster response time than nodes at higher indexes.
+     * @param List<InetAddressAndPort> endpoints.
+     * @return void.
+     */
+    public default void sortEndpointsByResponseTime(List<InetAddressAndPort> endpoints)
```

Comment:
@maxim-chn do this mean that if I call this method, it will throw an exception because `isEpAResponseTimeSlowerThanEpB` is not implemented? So by default it will throw exception?

Reply:
@smiklosovic , yes indeed.
The motivation is to encourage any future implementation of `IFailureDetector` (besides `FailureDetector`) to take into consideration how 2 endpoints are compared in terms of response time.